derive_builder_core/src/block.rs

```diff
@@ -55,26 +62,47 @@ impl FromStr for Block {
     /// opening/closing delimiters like `{`, `(` and `[` will be _rejected_ as
     /// parsing error.
     fn from_str(expr: &str) -> Result<Self, Self::Err> {
-        let b: syn::Block =
-            syn::parse_str(&format!("{{{}}}", expr)).map_err(|e| format!("{}", e))?;
-        Ok(Self(b.stmts))
+        syn::parse_str(&format!("{{{}}}", expr)).map(Self)
+    }
+}
+
+impl TryFrom<&'_ LitStr> for BlockContents {
+    type Error = syn::Error;
+
+    fn try_from(s: &LitStr) -> Result<Self, Self::Error> {
+        let mut block_str = s.value();
+        block_str.insert(0, '{');
+        block_str.push('}');
+        let block = syn::LitStr::new(&block_str, s.span()).parse::<Block>()?;
+        Ok(Self(block))
+    }
+}
+
+impl From<syn::Expr> for BlockContents {
```

Comment:
Is this needed outside of testing?

Reply:
No, but the tests that use it are far enough apart as to be annoying to have this anywhere else.